File: client/src/modules/models/Car.js
/**
 * Created by CPU60133_LOCAL on 12/6/2017.
 */

var Car = cc.Class.extend({
    
    deliveryPrice: 0,
    deliveryExp: 0,

    status: 0,

    ctor: function (deliveryPrice, deliveryExp) {
        //this.deliveryPrice = deliveryPrice;
        //this.deliveryExp = deliveryExp;

        this.updateDelivery(deliveryPrice, deliveryExp);
    },
    
    getStatus: function () {
        return this.status;
    },


    updateDelivery: function (price, exp) {
        this.deliveryPrice = price;
        this.deliveryExp = exp;

        if (this.deliveryPrice > 0) {
            this.status = DeliveryStatus.RECEIVABLE;
        } else {
            this.status = DeliveryStatus.EMPTY;
        }
    },

    receive: function () {
        if (this.getStatus() == DeliveryStatus.RECEIVABLE){
            user.addGold(this.deliveryPrice);
            user.addExp(this.deliveryExp);

            //
            this.updateDelivery(0, 0);
            return true;
        }
        return false;
    }
    
});

File: server/src/model/Car.java
package model;

import config.enums.DeliveryStatus;
import config.enums.ErrorLog;

public class Car {
    
    private int deliveryPrice;
    private int deliveryExp;
    
    private short status;
    
    public Car() {
        super();
        
        this.updateDelivery(0, 0);
    }


    public void setDeliveryPrice(int deliveryPrice) {
        this.deliveryPrice = deliveryPrice;
    }

    public int getDeliveryPrice() {
        return deliveryPrice;
    }

    public void setDeliveryExp(int deliveryExp) {
        this.deliveryExp = deliveryExp;
    }

    public int getDeliveryExp() {
        return deliveryExp;
    }

    public void setStatus(short status) {
        this.status = status;
    }

    public short getStatus() {
        return status;
    }



    public void updateDelivery(int deliveryPrice, int deliveryExp){
        this.deliveryPrice = deliveryPrice;
        this.deliveryExp = deliveryExp;
        
        if (this.deliveryPrice > 0) {
            this.status = DeliveryStatus.RECEIVABLE.getValue();
        } else {
            this.status = DeliveryStatus.EMPTY.getValue();
        }
    }

    //
    public short receive(ZPUserInfo user){
        if (this.getStatus() == DeliveryStatus.RECEIVABLE.getValue()){
            user.addGold(this.deliveryPrice);
            user.addExp(this.deliveryExp);
            //
            this.updateDelivery(0, 0);
            
            return ErrorLog.SUCCESS.getValue();
        }
        return ErrorLog.ERROR_ORDER_NOT_COMPLETE.getValue();
    }

}
